test/integration_test.go

```diff
@@ -2401,6 +2403,30 @@ func TestAWSGetCallerIdentityFunctions(t *testing.T) {
 	assert.Equal(t, outputs["user_id"].Value, *identity.UserId)
 }
 
+func TestGetPlatform(t *testing.T) {
+	t.Parallel()
+
+	cleanupTerraformFolder(t, TEST_FIXTURE_GET_PLATFORM)
+	tmpEnvPath := copyEnvironment(t, TEST_FIXTURE_GET_PLATFORM)
+	rootPath := util.JoinPath(tmpEnvPath, TEST_FIXTURE_GET_PLATFORM)
+
+	runTerragrunt(t, fmt.Sprintf("terragrunt apply-all --terragrunt-non-interactive --terragrunt-working-dir %s", rootPath))
+
+	// verify expected outputs are not empty
+	stdout := bytes.Buffer{}
+	stderr := bytes.Buffer{}
+
+	require.NoError(
+		t,
+		runTerragruntCommand(t, fmt.Sprintf("terragrunt output -no-color -json --terragrunt-non-interactive --terragrunt-working-dir %s", rootPath), &stdout, &stderr),
+	)
+
+	outputs := map[string]TerraformOutput{}
+
+	require.NoError(t, json.Unmarshal([]byte(stdout.String()), &outputs))
+	assert.Equal(t, outputs["platform"].Value, runtime.GOOS)
```

Comment:
Much better test, thanks 👍 

NIT: To ensure we don't get a panic if this test uncovers a bug, probably safer to do:

```suggestion
	platform, hasPlatform := outputs["platform"]
	require.True(t, hasPlatform)
	require.Equal(t, platform.Value, runtime.GOOS)
```

Reply:
TIL: what NIT means! 😁